ouroboros-consensus-byron-test/src/Test/Consensus/Byron/Generators.hs

```diff
@@ -272,6 +278,28 @@ instance Arbitrary ByronTransition where
 instance Arbitrary (LedgerState ByronBlock EmptyMK) where
   arbitrary = ByronLedgerState <$> arbitrary <*> arbitrary <*> arbitrary
 
+-- | Generator for a Byron ledger state in which the tip of the ledger given by
+-- `byronLedgerTipBlockNo` matches that derived from the `ChainValidationState` in
+-- `byronLedgerState`.
+genByronLedgerState :: Gen (LedgerState ByronBlock EmptyMK)
```

Comment:
While it makes sense that the tip block number matches the chain state, I wonder if we should generate ledger states in which this is not the case. Don't we want to test our system under these circumstances?

Reply:
The translation then fails; my assumption that "invalid" ledger states are only due to programming errors, in which case I don't think we should test the behaviour, but that assumption might be wrong.